superstruct/src/__tests__/superstruct.ts

```diff
@@ -1,45 +1,72 @@
-import { object, number, string, boolean, array, optional } from 'superstruct';
+import {
+  object,
+  number,
+  string,
+  optional,
+  pattern,
+  size,
+  union,
+  min,
+  max,
+  Infer,
+  define,
+  array,
+  boolean,
+} from 'superstruct';
 import { superstructResolver } from '..';
 
-const Article = object({
-  id: number(),
-  title: string(),
-  isPublished: optional(boolean()),
+const Password = define('Password', (value, ctx) =>
+  value === ctx.branch[0].password);
+
+const schema = object({
```

Comment:
Honestly, I need to read those schemas doc more, but you are certainly an expert now with all the schemas. ❤️